### DISCUSSION TITLE
Correct arguments for the callback for OnFinish event

### DISCUSSION THREAD

**Discussion by fakharak** [swoole/swoole-src#4257] [Q&A] [answered]
On https://www.swoole.co.uk/docs/modules/swoole-server-sendmessage
callback takes three arguments, as below:

```
$server->on('finish', function ($server, $fd, $from_id)
{

});
```

But, on https://www.swoole.co.uk/docs/modules/swoole-server-task

callback takes entirely different arguments, as:

```
$server->on('Finish', function (Swoole\Server $server, $task_id, $data)
{
    echo "Task#$task_id finished, data_len=" . strlen($data) . PHP_EOL;
});
```

----

Exactly how many arguments and which arguments can i use for callback for onFinish event ?

**Comment by sy-records** [comment, marked as answer]
```php
function onFinish(Swoole\Server $server, int $task_id, mixed $data)
```

**Reply by fakharak** [reply]
Thank you, that helps.
